### PR TITLE
Contributor guide folder and central starting point README outline

### PR DESCRIPTION
**Pull request by guineveresaenger**
This is the starting point for creating a single source of truth for how to contribute to Kubernetes.

A follow-up umbrella issue will be filed that contains a list of tasks that need to be completed to fully  migrate all the information to a central source of truth.

The README itself contains references to tasks that are still outstanding, but it would be helpful to have it merged into master, so we have somewhere to point people, to make them aware this is being worked on, and perhaps to garner more help.

### CODE REVIEW

**Comment by cblecker** [comment]
/ok-to-test
/cc
/sig contributor-experience

Will review this :)

File: contributors/guide/README.md [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+
+_Improvements needed_
+
+* The top level k/community/README.md should be a good starting point for what the community is and does. (see above instructions on rewriting this file)
+
+## Events
+
+Kubernetes is the main focus of CloudNativeCon/KubeCon, held every spring in Europe and winter in North America. Information about these and other community events is available on the CNCF [events](https://www.cncf.io/events/) pages.
+
+### Meetups
```
Comment (idvoretskyi) [inline contributors/guide/README.md]: Please, leave as is, I'll update after merging.

File: contributors/guide/README.md [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+
+__Improvements needed_
+* include link to meetups
+* information on CNCF support for founding a Meetup
+
+### Kubecon
```
Comment (idvoretskyi) [inline contributors/guide/README.md]: KubeCon

File: contributors/guide/README.md [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+
+If you haven't noticed by now, we have a large, lively, and friendly open-source community. We depend on new people becoming members and regular code contributors, so we would like you to come join us. To find out more about our community structure, different levels of membership and code contributors, please [explore here](https://github.com/kubernetes/community/blob/master/community-membership.md).
+
+_Improvements needed_
+
+* The top level k/community/README.md should be a good starting point for what the community is and does. (see above instructions on rewriting this file)
+
+## Events
+
+Kubernetes is the main focus of CloudNativeCon/KubeCon, held every spring in Europe and winter in North America. Information about these and other community events is available on the CNCF [events](https://www.cncf.io/events/) pages.
```
Comment (idvoretskyi) [inline contributors/guide/README.md]: Kubernetes is the main focus of CloudNativeCon/KubeCon, held twice per year in EMEA and in North America. Information about these and other community events is available on the CNCF [events](https://www.cncf.io/events/) pages.

Comment (idvoretskyi) [reply contributors/guide/README.md]: Edited. We don't necessarily run KubeCon only in winter and spring. Also, not necessarily Europe (but EMEA).

File: contributors/guide/README.md [hunk 4]
```diff
@@ -0,0 +1,212 @@
+**OWNER:**
+
+SIG-ContribX
```
Comment (cblecker) [inline contributors/guide/README.md]: nit: sig-contributor-experience

File: contributors/guide/README.md [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+
+
+# WARNING
```
Comment (cblecker) [inline contributors/guide/README.md]: Starting a new contributor document with an H1 WARNING isn't super friendly. Perhaps say this is a disclaimer?

File: contributors/guide/README.md [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+Hello! This is the starting point for our brand new contributor guide, currently underway as per [-issue#6102](https://github.com/kubernetes/website/issues/6102) and in need of help. Please be patient, or fix a section below that needs improvement, and submit a pull request!
+
+Many of the links below should lead to relevant documents scattered across the community repository. Often, the linked instructions need to be updated or cleaned up. 
+
+* If you do so, please port the relevant file from its previous location to come live in the community/contributors/guide folder, and delete its previous location.
```
Comment (cblecker) [inline contributors/guide/README.md]: suggestion: s/port/{move|relocate}/ -- easier to understand, especially with newer english speakers

File: contributors/guide/README.md [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+* Ultimately, all contributor guide specific files will live in this folder.
```
Comment (cblecker) [inline contributors/guide/README.md]: suggestion: s/Ultimately/Our goal is/ -- more open tone as it conveys that we're a big team and inviting people to become a part of it

File: contributors/guide/README.md [hunk 8]
```diff
@@ -0,0 +1,212 @@
...
+
+# Welcome
+
+Welcome to Kubernetes! This document is the single source of truth for how to contribute to the code base. Please leave comments / suggestions if you find something is missing or incorrect.
+
+# Before you get started
+
+## Sign the CLA
+
+Before you can contribute, you will need to sign the [Contributor License Agreement](https://github.com/kubernetes/community/blob/master/CLA.md).
```
Comment (cblecker) [inline contributors/guide/README.md]: Relative links work, and help when you're working in a fork or branch. So something like `/CLA.md` in this markdown file should link to the right place. You can do a find/replace for anything `https://github.com/kubernetes/community/blob/master/` or `https://github.com/kubernetes/community/tree/master/` and subsitute it with `/`.

File: contributors/guide/README.md [hunk 9]
```diff
@@ -0,0 +1,212 @@
...
+_Improvements needed_
+
+* The top level k/community/README.md should be a good starting point for what the community is and does. (see above instructions on rewriting this file)
+
+## Events
+Kubernetes is the main focus of CloudNativeCon/KubeCon, held twice per year in EMEA and in North America. Information about these and other community events is available on the CNCF [events](https://www.cncf.io/events/) pages.
+
+### Meetups
+
+__Improvements needed_
```
Comment (cblecker) [inline contributors/guide/README.md]: I think the two underscores here mess up the formatting a bit.

File: contributors/guide/README.md [hunk 10]
```diff
@@ -0,0 +1,212 @@
...
+
+Please make sure to read and observe our [Code of Conduct](https://github.com/cncf/foundation/blob/master/code-of-conduct.md)
+
+### Code Review
+
+For a brief description of the importance of code review, please read [On Code Review](https://github.com/kubernetes/community/blob/master/contributors/devel/community-expectations.md#code-review).
+
+_Improvements needed_
+* edit above link to only provide code review section.
+* decide whether this is something that belongs in this section, or should be combined with Contributing instructions [below](https://github.com/guineveresaenger/community/tree/contributor-guide/contributors/guide#contributing).
```
Comment (cblecker) [inline contributors/guide/README.md]: Links to a personal repo, per above, this should be a relative link (e.g. `/contributor-guide/contributors/guide#contributing`)

File: contributors/guide/README.md [hunk 11]
```diff
@@ -0,0 +1,212 @@
...
+
+* Make people aware of “extra” contributing guidelines on a SIG.
+
+* At the same time, file issues with all the SIGs to not have duplicate things in their CONTRIBUTING.md. Keep it light, keep it clean, have only one source of truth. 
+
+* Do not overload people with the entirety of the SIG structure; it is very intimidating to a newcomer
+
+* Possible text to elaborate on (the sig list should remain in its current place however):
+
+    * ([Pick a SIG](https://github.com/kubernetes/community/blob/master/sig-list.md), peruse its associated [cmd](https://github.com/kubernetes/kubernetes/tree/master/cmd) directory, find a main() and read code until you find something you want to fix.)
```
Comment (cblecker) [inline contributors/guide/README.md]: git.k8s.io/kubernetes/master/cmd

**Comment by guineveresaenger** [comment]
Thank you @cblecker I will update tomorrow PST :)

File: contributors/guide/README.md (hunk 5 above)
Comment (idvoretskyi) [reply contributors/guide/README.md]: +1

File: contributors/guide/README.md (hunk 8 above)
Comment (idvoretskyi) [reply contributors/guide/README.md]: I'd keep the full path if this document will be moved somewhere.

Comment (cblecker) [reply contributors/guide/README.md]: Oh, if the goal is to move this out of community, then I agree.

In that case, I’d suggest https://git.k8s.io/community/CLA.md (and similar). That’s a redirect we own and provides for a clean, consistent anchor for fully qualified URLs

Comment (guineveresaenger) [reply contributors/guide/README.md]: Huh. I was unaware of those. Do all kubernetes github files have such a redirect?

And yes, relative files +1, but _after_ we move them into the /guide folder, which in the case of CLA we should.

File: contributors/guide/README.md (hunk 10 above)
Comment (guineveresaenger) [reply contributors/guide/README.md]: whoops

File: contributors/guide/README.md (hunk 8 above)
Comment (cblecker) [reply contributors/guide/README.md]: git.k8s.io is just an nginx redirector.. code is here: https://github.com/kubernetes/k8s.io/blob/8320db3841550deb6084137e0533fdc8c193d9b7/k8s.io/configmap-nginx.yaml#L183-L193

and yes, it works for any github.com/kubernetes/ repo

**Comment by guineveresaenger** [comment]
Apologies @cblecker - got a bit swamped on Friday afternoon. All changes should be addressed now.

**Comment by cblecker** [comment]
@stp-ip @guineveresaenger In future, we should be squashing before merge (to keep the tree at least somewhat clean)

**Comment by stp-ip** [comment]
@cblecker Yes missed that. Was checking the whole content and if all comments were addressed. Sorry my fault.

**Comment by guineveresaenger** [comment]
@cblecker in kubernetes do we "squash and merge" or do we squash before submitting PR/force-push with altered history in case of changes?

**Comment by guineveresaenger** [comment]
(I had assumed the former)

**Comment by cblecker** [comment]
The latter. The actual merge is handled by a bot, and just does a straight git merge

**Comment by cblecker** [comment]
Having split up commits helps from a review standpoint, but then once all comments are addressed, usually the last step is the author squashing into a readable git history

**Comment by guineveresaenger** [comment]
Makes sense. Thank you for your patience. _makes mental note to go through that part of this very documentation with a fine comb_

**Pull request merged**